templates/account/js/metadata.js

```diff
@@ -1,8 +1,88 @@
 <script>
-  $("#edit-button").on('click', function(){
-    $("#input-form").removeClass("hidden");
-    $("#edit-button").hide();
-    $("#no-info-text").hide();
-    $("#user-info").hide();
-  })
+let country_name_to_country_code = new Map(Object.entries({{ country_name_to_country_code }}))
+let country_code_to_country_name = new Map(Object.entries({{ country_code_to_country_name }}))
+
+// hide location dropdown, use country code and country name instead
+$("#input-form > form > div:nth-child(3)").hide();
+$('#locations').val([]);
+
+$("#edit-button").on('click', function () {
+  $("#input-form").removeClass("hidden");
+  $("#edit-button").hide();
+  $("#no-info-text").hide();
+  $("#user-info").hide();
+});
+
+let previous_ccs = get_val_or_default('#country_codes', []);
+let previous_cns = get_val_or_default('#country_names', []);
+
+$('#country_codes').change(function () {
```

Comment:
Consider refactoring to a common function for the code inside `$('#country_codes').change()` and `$('#country_names').change()`. This would help in code re-use. _Optional._

```js
const onChangeCountry = (current, previous, selector) => {
   ...
};

$('#country_codes').change(() => { onChangeCountry(current_ccs, previous_ccs, '#country_names') });
$('#country_names').change(() => { onChangeCountry(current_cns, previous_cns, '#country_codes') });
```

Reply:
great idea! refactored via 8201a03